src/lib/api/index.ts

```diff
@@ -244,54 +253,43 @@ export async function post(agent: BskyAgent, opts: PostOpts) {
 
   try {
     // TODO: this needs to be batch-created with the post!
-    if (opts.threadgate?.length) {
-      await createThreadgate(agent, res.uri, opts.threadgate)
-    }
+    await writeThreadgateRecord({
+      agent,
+      postUri: res.uri,
+      threadgate: createThreadgateRecord({
+        post: res.uri,
+        allow: threadgateAllowUISettingToAllowRecordValue(opts.threadgate),
+      }),
+    })
   } catch (e: any) {
-    console.error(`Failed to create threadgate: ${e.toString()}`)
+    logger.error(`Failed to create threadgate`, {
+      context: 'composer',
+      safeMessage: e.message,
+    })
     throw new Error(
-      'Post reply-controls failed to be set. Your post was created but anyone can reply to it.',
+      'Failed to save post interaction settings. Your post was created but users may be able to interact with it.',
```

Comment:
Would it be feasible to mark this error string for localization?

Reply:
Love that suggestion. I'll make a separate PR to clean some of this up and include these once this PR is in 👍

Reply:
@estrattonbailey Afaik the user-facing strings in `src/lib/api/index.ts` are some of the only ones in the app that can't yet be localized, so when you've got a moment sometime that PR would be appreciated :)